packages/headless/src/chat.ts

```diff
@@ -685,3 +703,86 @@ export function useMessages(room: number | null) {
     more,
   ]);
 }
+
+type RoomId = number;
+type UserId = number;
+
+export function useChatStatus() {
+  const socket = useSocket();
+
+  /**
+   * A map from rooms to object containing both users in it and showing
+   * if they are typing currently or not
+   * ```ts
+   * type TypingStaus = boolean; // true in case the user is typing.
+   * type TypingMap = Record<RoomId, Record<UserId, TypingStatus>>;
+   * ```
+   */
+  const [typingMap, setTypingMap] = useState<{
+    [roomId: number]: { [userId: number]: boolean };
```

Comment:
Because you will move the `RoomMap` type here in this file, this mean we can use it here.